src/element.ts

```diff
@@ -9,22 +9,42 @@
 import { Vex } from './vex';
 import { Registry } from './registry';
 import { Flow } from './tables';
+import { BoundingBox } from './boundingbox';
+import { Font } from './smufl';
+import { RenderContext, ElementStyle, ElementAttributes } from './types/common';
 
-export class Element {
-  static newID() {
-    return 'auto' + Element.ID++;
+export abstract class Element {
+  protected static ID: number = 1000;
+
+  protected context?: RenderContext;
+
+  protected rendered: boolean;
+
+  protected style?: ElementStyle;
+
+  protected attrs: ElementAttributes;
+
+  protected classes: Record<string, boolean>;
+
+  protected boundingBox?: BoundingBox;
+
+  protected fontStack: Font[];
+
+  protected musicFont: Font;
+
+  protected registry?: Registry;
+
+  static newID(): string {
+    return `auto${Element.ID++}`;
   }
 
-  constructor({ type } = {}) {
+  constructor() {
     this.attrs = {
       id: Element.newID(),
-      el: null,
```

Comment:
@0xfe I also removed `el` because it is not used in the code or the tests. Is this ok with you?

Reply:
This is used to associate DOM elements with vexflow elements. (We should have tests for it, but we don't unfortunately.)

Reply:
ok